src/nodes.coffee

```diff
@@ -980,6 +999,13 @@ exports.StringLiteral = class StringLiteral extends Literal
     copy.locationData = locationData
     copy
 
+  shouldGenerateTemplateLiteralAst: ->
+    @quote.length is 3
```

Comment:
I'd run into cases where a leading heredoc was being treated as a directive. I guess in theory someone could be using them this way (since in the generated JS the heredoc will just be a normal string) but it seems bizarre/unexpected to treat a heredoc as a directive

This relates to what we generate as AST for heredocs - where I landed was that we should generate `TemplateLiteral` AST nodes (ie JS `` `...` ``) for all heredocs (rather than `StringLiteral`). This seemed more appropriate to me since they're more similar to template literals with respect to things like newlines

Reply:
Yes, see #5019. It’s been a leftover TODO from the CoffeeScript 2 effort that we need to output heredoc strings as template literals using backticks.